src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -354,8 +357,29 @@ - (void)startWithWriteable:(id<GRXWriteable>)writeable {
   _retainSelf = self;
 
   _responseWriteable = [[GRXConcurrentWriteable alloc] initWithWriteable:writeable];
+
+  _wrappedCall = [[GRPCWrappedCall alloc] initWithHost:_host path:_path];
+  NSAssert(_wrappedCall, @"Error allocating RPC objects. Low memory?");
+
   [self sendHeaders:_requestHeaders];
   [self invokeCall];
+  // TODO(jcanizales): Extract this logic somewhere common.
+  NSString *host = [NSURL URLWithString:[@"https://" stringByAppendingString:_host]].host;
+  if (!host) {
+    // TODO(jcanizales): Check this on init.
+    [NSException raise:NSInvalidArgumentException format:@"host of %@ is nil", _host];
+  }
+  __weak typeof(self) weakSelf = self;
+  _connectivityMonitor = [GRPCConnectivityMonitor monitorWithHost:host];
```

Comment:
Looking a little more, it seems partially to be because `SCNetworkReachabilityCreateWithName` will attempt to resolve DNS  with `gethostbyname`... while `SCNetworkReachabilityCreateWithAddress` with `0.0.0.0` will just tell you if there is an up network interface